### ISSUE TITLE
List<dynamic>' is not a subtype of type 'PlaceDetails'

### ISSUE THREAD

**Issue by dereklakin** [lejard-h/google_maps_webservice#53]
`getDetailsByPlacedId` is throwing an exception: "Exception has occurred.
_TypeError (type 'List<dynamic>' is not a subtype of type 'PlaceDetails')" Stack trace shows it is on `PlaceDetails.fromJson` in this section of places.dart:

```
  factory PlacesDetailsResponse.fromJson(Map json) => json != null
      ? PlacesDetailsResponse(
          json['status'],
          json['error_message'],
          json['result'] != null ? PlaceDetails.fromJson(json['result']) : [],
          json['html_attributions'] != null
              ? (json['html_attributions'] as List)?.cast<String>()
              : [])
      : null;
```

**Comment by dereklakin** [comment]
Turns out I was using the `id` field for the `place_id`. I should (of course) be using `placeId`.

**Comment by yoyoyo11** [comment]
@dereklakin Hi. I got the same error but only when I specifically chose this place: "Marvel Studios LLC, Rosecrans Ave # 7-110, Manhattan Beach, CA, USA". It works for any other location I enter. I use placeId for .getDetailsByPlaceId so I'm not exactly sure what to do. Any guidance helps. Also, can you test the above address and let me know whether it threw an exception or not? Thanks!

**Comment by dereklakin** [comment]
> @dereklakin Hi. I got the same error but only when I specifically chose this place: "Marvel Studios LLC, Rosecrans Ave # 7-110, Manhattan Beach, CA, USA". It works for any other location I enter. I use placeId for .getDetailsByPlaceId so I'm not exactly sure what to do. Any guidance helps. Also, can you test the above address and let me know whether it threw an exception or not? Thanks!

Sorry for the delay. I can't find that place on Google Maps. Can you give me a URL to the place on Google Maps, please?

**Comment by yoyoyo11** [comment]
As luck would have it, the place I specified ("Marvel Studios LLC, Rosecrans Ave # 7-110, Manhattan Beach, CA, USA") is no longer "findable" in the google places field. I don't see it in the prediction result set while typing it out. I did a "sanity check" in Google Maps and it's not appearing either (street view shows that it's a corporate building that's out-of-business"). Perhaps that business was closing and the data had not yet propagated to Google Places? Is that even possible? I suppose this is a non-issue now but I expect these types of edge-cases to not appear in the future. Thanks.
